client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/navigator-buttons/index.tsx

```diff
@@ -44,10 +46,24 @@ const GenericButton = ( { icon, children, ...props }: Props ) => {
 	);
 };
 
-export const NavigationButtonAsItem = ( { className, ...props }: Props ) => {
+const ChecklistButton = ( { icon, children, className, checked, ...props }: Props ) => {
+	return (
+		<GenericButton
+			{ ...props }
+			icon={ checked ? check : icon }
+			className={ classnames( className, {
+				'navigator-button__checklist-item--checked': checked,
+			} ) }
+		>
+			{ children }
+		</GenericButton>
+	);
+};
+
+export const NavigationButtonAsItem = ( { className, checklist, ...props }: Props ) => {
 	return (
 		<NavigatorButton
-			as={ GenericButton }
+			as={ checklist ? ChecklistButton : GenericButton }
 			className={ classnames( 'navigator-button', className ) }
```

Comment:
We can simply pass the props you want here, and then we don't need the `ChecklistButton`